### ISSUE TITLE
New commit can not recognize gender

### ISSUE THREAD

**Issue by Zumbalamambo** [oarriaga/face_classification#17]
why is that the newest commit is not able to recognise the gender?

**Comment by oarriaga** [comment]
Hi,
Which demo are you running? I provided a new demo that only includes emotion classification and displays a color with the selected emotion (video_emotion_color_demo.py). However, the previous demo; that includes emotion and gender classification, is located in the file (video_emotion_gender_demo.py).

**Comment by Zumbalamambo** [comment]
Okay thank you.. May I know why is that the accuracy of gender is low? Will it increase if I retrain it?

**Comment by oarriaga** [comment]
The obtained accuracy for the gender classifiction is of 96% with the IMDB dataset. I believe that a new model could be built/trained so that can generalize better. Also another unanswered question is how biased is the network with respect to "celebrity faces". One specific example of this is the use of glasses. I have found empirically that woman wearing glasses are more prone to be classified by the network as man. A look into the dataset might be reveal that there are few cases within the data for which a woman is wearing glasses.

**Comment by Zumbalamambo** [comment]
can u please share with me the imdb pretrained model?

**Comment by oarriaga** [comment]
All pre-trained models are given in the directory "trained_models/".
The IMDB gender trained model is given in "trained_models/gender_models/"

**Comment by Zumbalamambo** [comment]
Is it possible to retrain the existing model? If so how the data set has to be ?

**Comment by oarriaga** [comment]
Yes, please refer to the README before asking. As explained in the README you only have to run
> python3 train_emotion_classifier.py

**Comment by Zumbalamambo** [comment]
Thank you.. yes I read it...  after collecting images, where should I place the datas for train_emotion_classifier to pick?

**Comment by AlainPilon** [comment]
I am also experiencing some issues with gender identification, in both directions. The most notorious of these case being a guy with a moustache and glasses identified as a woman :-O

**Comment by oarriaga** [comment]
Gender identification is highly biased towards the dataset in which it was trained (IMDB dataset). Moreover,  there is a variable in the code called `gender_offsets` which make the image crop bigger or smaller than the one given by the detection model. You can change this variable to 0 so that it does not add anything to the frame but the face. This might alleviate the problem.

**Comment by AlainPilon** [comment]
Thanks for the prompt feedback. 

I will spend more time working on this for sure later this week. 

I think the project is great and really hope I can provide some PR to make it easier to use/understand.
